app/controllers/submissions_controller.rb

```diff
@@ -47,30 +43,30 @@ def authenticate_to_submit
   end
 
   def posts
-    @posts = Submission.posts
+    @submissions ||= Submission.posts
   end
 
   def post
-    @post = Submission.find(params[:id])
+    @submission ||= Submission.find(params[:id])
   end
 
-  def new_post
-    @post = Submission.new
+  def new_submission
+    @submission ||= Submission.new
   end
 
   def new_comment
-    @comment = Submission.new
+    @comment ||= Submission.new
```

Comment:
I realize I may have said otherwise previously (I honestly can't remember, so sorry if I did), but I wouldn't expect this method to memoize an instance variable. A method prefixed with `new_` should probably return a new instance every time it's invoked.

```ruby
def new_comment
  Submission.new
end
```

Does that make sense?

Reply:
"A method prefixed with new_ should probably return a new instance every time it's invoked." Yes, that does make sense.... the memoization was first suggested for `new_comment` - https://github.com/gingin77/hackernews_clone/pull/5#pullrequestreview-94148505

Reply:
Yea, I'm sure I just commented on the last method I saw with this pattern without thinking too deeply about it. My b.